src/core/options-container.cpp

```diff
@@ -27,17 +27,47 @@ void options_container::update( std::shared_ptr< extension_snapshot > ext )
     auto ctr = As< options_container >( ext );
     if( ! ctr )
         return;
-    for( auto && opt : ctr->_options )
+    for( auto id : ctr->_ordered_options )
```

Comment:
Will this affect USB devices ordering?

Reply:
It affects all devices.
But librealsense returns the device options already ordered for USB, so no change.

Reply:
Hmm maybe I misunderstood?
Just call me and we'll talk if it's unclear.

Reply:
Main question is if on D400 USB the UI remain the same with the same order?

Reply:
Yes, it should:

`supported_options` (what device-model.cpp used to iterate over) is really `sub->options_metadata' which is an option-ID-based map.

`options_metadata` is populated in `subdevice_model` constructors using `supported_options = s->get_supported_options()` where s is a `sensor`.

And `rs2::options::get_supported_options()` calls `rs2_get_options_list()` which calls `options_container::get_supported_options()` which you can see in line 64, still returns options ordered by the option ID.

It's only in SW devices that we override `get_supported_options()` to return the options in their registeration order.